### DISCUSSION TITLE
Private dependencies not being looked up

### DISCUSSION THREAD

**Discussion by ramraaj25** [renovatebot/renovate#17369] [Suggest an Idea]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.163.1

### Please select which platform you are using if self-hosting.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I have a go application hosted on my org's bitbucket cloud server where I have some private dependencies which are also hosted on my org's bitbucket cloud server but Renovate doesn't seem to be able to detect these private repo upgrades. It only creates PRs for dependencies which are public.  
Why is this happening ? I mean if Renovate can access the bitbucket server and pull my working repo from it why can't it scan the other repos inside the same for available updates ?
My config.js looks something like this: 
```
module.exports = {
    endpoint: "https://bitbucket.myorg.com/",
    username: "myusername",
    password: "mypassword",
    repositories: [
        { repository: 'project/repo', baseBranches: ["mybranch"] }
    ]
    platform: 'bitbucket-server',
    onboardingConfig: { extends: ["config:base", ":prHourlyLimitNone"] },
    hostRules: [
        {
          matchHost: "github.com",
          token: "my_token"
        },
        {
          matchHost: "bitbucket.myorg.com",
          username: 'myusername',
          password: 'mypassword'
        },
    ],
    golang: {
      GOPRIVATE: "bitbucket.myorg.com",
      GOPROXY: "https://proxy.golang.org,direct",
    },
};
```

### Relevant debug logs

<details><summary>The log from the onboarding PR:  </summary>

```
⚠ Dependency Lookup Warnings ⚠
Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.

Failed to look up dependency bitbucket.myorg.com/project/repo1
Failed to look up dependency bitbucket.myorg.com/project/repo2
Failed to look up dependency bitbucket.myorg.com/project/repo3
Files affected: go.mod
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by MaronHatoum** [comment]
Hi @ramraaj25 

I think `hostType: "git-tags"` is missing, 

you could try using the following `hostRules`:

```
module.exports = {
    endpoint: "https://bitbucket.myorg.com/",
    username: "myusername",
    password: "mypassword",
    repositories: [
        { repository: 'project/repo', baseBranches: ["mybranch"] }
    ]
    platform: 'bitbucket-server',
    onboardingConfig: { extends: ["config:base", ":prHourlyLimitNone"] },

    hostRules: [
        {
          matchHost: "github.com",
          token: "my_token"
        },
        {
	  hostType: "git-tags",
          matchHost: "https://bitbucket.myorg.com",
          username: 'myusername',
          password: 'mypassword'
        },
    ],
    golang: {
      GOPRIVATE: "bitbucket.myorg.com",
      GOPROXY: "https://proxy.golang.org,direct",
    },
};
```

**Reply by rarkins** [reply]
Go uses the `go` datasource, not `git-tags`.

**Comment by rarkins** [comment]
I don't think that Bitbucket Cloud is supported as a source for the `go` datasource today. This could be verified by testing out with a public module hosted on Bitbucket Cloud.

If public modules on bitbucket cloud do not work as I'm suspecting, it means we need a feature request and a reproduction repo can be easily set up (no credentials are relevant).

If public modules work but private do not, then we need a feature request for private module support on bitbucket cloud. A reproduction could also be set up, but not as easily.

**Reply by ramraaj25** [reply]
Thanks for the reply.
A Correction: I figured out that my org uses On-Prem Bitbucket Server and not Bitbucket Cloud Server.

Moreover, I tried importing and using a public repo hosted on the same and Renovate failed to lookup dependencies for it as well just as you suspected.
